iothub_client/samples/pnp/common/pnp_device_client_helpers.h

```diff
@@ -6,16 +6,61 @@
 
 #include "iothub_device_client.h"
 
+//
+// Whether we're using a connection string or DPS provisioning for device credentials
+//
+typedef enum PNP_CONNECTION_SECURITY_TYPE_TAG
+{
+    PNP_CONNECTION_SECURITY_TYPE_CONNECTION_STRING,
+    PNP_CONNECTION_SECURITY_TYPE_DPS
+} PNP_CONNECTION_SECURITY_TYPE;
+
+#ifdef USE_PROV_MODULE
+//
+// PNP_DPS_CONFIGURATION is used to configure the DPS device client
+//
+typedef struct PNP_DPS_CONFIGURATION_TAG
+{
+    char* idScope;
+    char* registrationId;
+    char* deviceKey;
+} PNP_DPS_CONFIGURATION;
+#endif /* USE_PROV_MODULE */
+
+//
+// PNP_HELPER_DEVICE_CONFIGURATION is used to setup the IOTHUB_DEVICE_CLIENT_HANDLE
+//
+typedef struct PNP_DEVICE_CONFIGURATION_TAG
+{
+    // Whether we're using connection string or DPS
+    PNP_CONNECTION_SECURITY_TYPE securityType;
+    // The connection string or DPS security information
+    union {
+        char* connectionString;
+#ifdef USE_PROV_MODULE
+        PNP_DPS_CONFIGURATION dpsConfiguration;
+#endif
+    } u;
```

Comment:
nit: call it something like `auth` or `connection_auth`?

Reply:
Fixing.